cadquery/occ_impl/shapes.py

```diff
@@ -2260,6 +2260,9 @@ def makeFromWires(cls, outerWire: Wire, innerWires: List[Wire] = []) -> "Face":
         Makes a planar face from one or more wires
         """
 
+        if innerWires and not outerWire.IsClosed():
```

Comment:
Why not just `if not outerWires.IsClosed()`?

Reply:
I see a remark about tests relaying on this, could you give more details?

Reply:
I wanted to avoid causing any existing tests to fail without more discussion.  This current commit prevents the segfault and existing tests still pass.

If `Face.makeFromWires` must enforce that all wires are closed, this commit is only an intermediate step in that direction.  For now, I did not change any existing tests.

Here are the details of the impacted tests that rely on open outer wires:

```
FAILED test_cadquery.py::TestCadQuery::test2Dchamfer - ValueError: Cannot build face(s): outer wire is not closed
FAILED test_cadquery.py::TestCadQuery::test2Dfillet - ValueError: Cannot build face(s): outer wire is not closed
FAILED test_cadquery.py::TestCadQuery::testParametricCurve - ValueError: Cannot build face(s): outer wire is not closed
FAILED test_cadquery.py::TestCadQuery::testSpline - ValueError: Cannot build face(s): outer wire is not closed
FAILED test_cadquery.py::TestCadQuery::testSplineShape - ValueError: Cannot build face(s): outer wire is not closed
FAILED test_sketch.py::test_face_interface - ValueError: Cannot build face(s): outer wire is not closed
```

### test2Dchamfer
https://github.com/CadQuery/cadquery/blob/9ef08d6872fc17f1ead3ce3c945d20ae9f8b1bd0/tests/test_cadquery.py#L4777-L4780

![2Dchamfer1](https://user-images.githubusercontent.com/16394272/148319514-76e0a4aa-2457-4715-891f-f514c7c77cb7.png)

The test could be updated to close the wire before the `chamfer2D` call.

![2Dchamfer2](https://user-images.githubusercontent.com/16394272/148318471-0fdf9b32-7266-456d-ad0f-3fed13875631.png)

### test2Dfillet

Similar to the `chamfer2D` test.

https://github.com/CadQuery/cadquery/blob/9ef08d6872fc17f1ead3ce3c945d20ae9f8b1bd0/tests/test_cadquery.py#L4756-L4760

![2Dfillet](https://user-images.githubusercontent.com/16394272/148318505-a3f8ae45-0d9c-4d02-b10e-433626e351a2.png)

### testParametricCurve

https://github.com/CadQuery/cadquery/blob/9ef08d6872fc17f1ead3ce3c945d20ae9f8b1bd0/tests/test_cadquery.py#L3657

![parametricCurve](https://user-images.githubusercontent.com/16394272/148318558-b4fcd477-782b-467b-a0ae-3eda114fc435.png)

### testSpline

Actually this test intentionally builds an invalid face.  It checks `isValid()` rather than for `ValueError` exception.

https://github.com/CadQuery/cadquery/blob/9ef08d6872fc17f1ead3ce3c945d20ae9f8b1bd0/tests/test_cadquery.py#L645-L648


### testSplineShape

Similar to `testParametricCurve`, this extrudes an open wire.  It calls `close()` and I don't see the spline so this test may need an update in any case.  I don't know that the intention was to extrude an open wire here.

https://github.com/CadQuery/cadquery/blob/9ef08d6872fc17f1ead3ce3c945d20ae9f8b1bd0/tests/test_cadquery.py#L1986-L1987

![splineshape](https://user-images.githubusercontent.com/16394272/148318578-468ba0e2-9a04-4436-8d3b-f3c03c19844f.png)

### test_face_interface

An additional point `(0,0)` would be required to close the polygon.

https://github.com/CadQuery/cadquery/blob/9ef08d6872fc17f1ead3ce3c945d20ae9f8b1bd0/tests/test_sketch.py#L45

```
>       s7 = Sketch().polygon([(0, 0), (0, 1), (1, 0)])

test_sketch.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../cadquery/sketch.py:310: in polygon
    return self.face(w, angle, mode, tag)
../cadquery/sketch.py:153: in face
    res = Face.makeFromWires(b)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
```

Reply:
Thanks @lorenzncode ! Very clear answer. I agree with your proposal and we can investigate further in another issue.